### ISSUE TITLE
no-use-before-declare false positive for undefined

### ISSUE THREAD

**Issue by buu700** [palantir/tslint#2442]
### Bug Report

This rule seems to incorrectly raise an error on every line where `undefined` is used in our code.

- __TSLint version__: `5.0.0`
- __TypeScript version__: `2.2.2`
- __Running TSLint via__: CLI

#### TypeScript code being linted

```ts
const balls = undefined;
```

with `tslint.json` configuration:

```json
{
	"rules": {
		"no-use-before-declare": true
	}
}
```

#### Actual behavior

`variable 'undefined' used before declaration`

#### Expected behavior

No error, as with tslint 4.x.

**Comment by nchen63** [comment]
duplicate of #2434
